src/jit/lsrabuild.cpp

```diff
@@ -1595,9 +1580,31 @@ void LinearScan::buildRefPositionsForNode(GenTree* tree, BasicBlock* block, Lsra
 
         assert((candidates & allRegs(srcInterval->registerType)) != 0);
 
-        // For non-localVar uses we record nothing, as nothing needs to be written back to the tree.
-        GenTree* const refPosNode = srcInterval->isLocalVar ? useNode : nullptr;
-        RefPosition*   pos        = newRefPosition(srcInterval, currentLoc, RefTypeUse, refPosNode, candidates, 0);
+        GenTree* refPosNode;
+        if (srcInterval->isLocalVar)
+        {
+            // We have only approximate last-use information at this point.  This is because the
```

Comment:
conditions before were:
` if (tree->OperIs(GT_LCL_VAR, GT_LCL_FLD)) and if (isCandidateVar(varDsc))`
now we have:
`if (srcInterval->isLocalVar) `
are they equal?

Reply:
Yes, they are equivalent- at this point we already have `srcInterval`, where before (when looking at the lclVar itself, as opposed to its use), we don't yet have the `Interval` available.